packages/sync/src/conflicts/strategies.ts

```diff
@@ -1,19 +1,9 @@
-import deepmerge from "deepmerge";
+import { ConflictResolutionData } from "./ConflictResolutionData";
 
-const diffMergeClientWins = (server: ConflictResolutionData, client: ConflictResolutionData) => {
-    return deepmerge(server, client);
+// Used as default strategy for SDK
+export const diffMergeClientWins = (server: ConflictResolutionData, client: ConflictResolutionData) => {
```

Comment:
Looks like this should have the arguments `(operationName: string, server: ConflictResolutionData, client: ConflictResolutionData)`

Because in `conflictLink.ts` you are calling `resolvedConflict = config.conflictStrategy(operation.operationName, data.serverData, data.clientData);`

I'm seeing a bug right now where when you create a conflict in the ionic example app, the `resolvedConflict` data coming back from `conflictLink.ts` is the server data instead of the client data and then it's doing a `nextState` on the server data which bumps the version to a higher version than on the server. This creates another conflict and it never gets resolved in the app.

Reply:
WOW. How this is even possible!
TypeScript oh my TypeScript so even you have betrayed me ;)